docs/challenges/language-support/csharp.mdx

```diff
@@ -0,0 +1,112 @@
+---
+title: "C#"
+description: "Learn how to solve CodeCrafters challenges in C#"
+---
+
+# Local development
+
+To run the code for your challenge locally, you'll need `dotnet` installed. Our test runners use Dotnet 8.0 (as of April 2024).
+
+The script for your challenge (like `./your_sqlite3.sh`) will automatically compile your code using `dotnet` before executing it. It'll
+look something like this:
+
+```sh
+#!/bin/sh
+
+exec dotnet run --project . --configuration Release "$@"
+```
+
+# File structure
+
+- The files for your solution are placed in the `src/` directory.
+- `src/Program.cs` contains the `main` function, which is what the test runner executes.
+- The `.csproj` file defines project settings, target frameworks and dependencies.
+
+# Adding more files
+
+You can add more files and directories to `src/`. The test runner will include them when compiling your code.
+
+For example, if you added a file at `src/Foo.cs`, you could use it like so:
+
+<CodeGroup>
+```csharp src/Foo.cs
+public class Foo {
+    public static void foo() {
+        System.out.println("Hello world!");
+    }
+}
+```
+
+```csharp src/Program.cs
+public class Main {
+    public static void Main(String[] args) {
+        Foo.foo();  // calls the method foo() from Foo class in Foo.cs
+    }
+}
+```
+
+</CodeGroup>
+
+By default, all classes in `src/` belong to the same namespace. This namespace is called the _project namespace_. E.g. If the project name is `MyProject`, all classes are defined under the `MyProject` namespace. This is why you can call `foo()` without importing anything.
+
+If you want to define a class under a _different_ namespace, you'll need to enclose it within braces like this:
+
+```csharp
+namespace AnotherNamespace {
+    public class Foo {
+        public static void foo(string[] args) {
+            // write code here
+        }
+    }
+}
+```
+
+Let's say you want to define the `Foo` class in `Foo.cs` under the `AnotherNamespace` namespace.
+
+<CodeGroup>
+
+```csharp src/Foo.cs
+namespace AnotherNamespace {
+    public class Foo {
+        public static void foo() {
+            System.out.println("Hello world!");
+        }
+    }
+}
+```
+
+```csharp src/Program.cs
+using AnotherNamespace
+
+public class Main {
+    public static void main(String[] args) {
+      Foo.foo();  // Uses the foo function from Foo class defined under AnotherNamespace
+    }
+}
+```
+
+</CodeGroup>
+
+<Note>
```

Comment:
This can be removed - seems like a more nuanced detail that we don't need to cover.